BenchPress/Generators.Tests/AzureDeploymentImporterTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Generators.LanguageProviders;
+
+namespace Generators.Tests
+{
+    [TestClass]
+    public class AzureDeploymentImporterTests
+  {
+      [TestMethod]
+      public void Should_Throw_If_Invalid_File_Format()
+      {
+          // arrange
+          var inputFilePath = "in.txt";
+          var outputFilePath = "outFolder";
+
+          // act
+          Action action = () => { AzureDeploymentImporter.Import(inputFilePath, outputFilePath); };
+
+          // assert
+          Assert.ThrowsException<FileFormatException>(action);
+      }
+
+      [TestMethod]
+      public void Should_Succeed_With_Bicep_File()
+      {
+          // arrange
+          var inputFilePath = "./DeploymentFiles/resourceGroup.bicep";
+          var outputFilePath = "outFolder";
+
+          // act
+          var result = AzureDeploymentImporter.Import(inputFilePath, outputFilePath);
```

Comment:
Typically, I'd be opposed to a test like this. Because this test actually reading/writing from/to the file system it is problematic. However, it fits our goal of putting a baseline in place, and it would take a significant refactoring of the code and/or test to make it more testable.

With that said, it might be worth looking into how to mock static methods such as `Bicep.Cli.Program.Main`, `File.ReadAllText`, and `File.Delete`. This is easy in Java, but in C# (given the 10 minutes I spent looking into it), it doesn't appear to be as simple. I did find references to Fakes that might be a good starting point: https://learn.microsoft.com/en-us/visualstudio/test/isolating-code-under-test-with-microsoft-fakes?view=vs-2022&tabs=csharp

Reply:
Yeah exactly, this was what I was saying earlier this morning about the code not being very testable atm. To my knowledge mocking static functions isn't possible without refactoring and adding some sort of dependency injections to mock static a function implementation.

There is a library called `System.IO.Abstraction` I believe that can help with mocking IO functions.

Reply:
Since `Microsoft.Fakes` requires VS Enterprise to use, we can't incorporate it into our CI/CD pipelines in Github. So reverting this. The solution would probably be to refactor the code to make it more unit testable.